OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java

```diff
@@ -202,9 +204,32 @@ public AlarmInfo getMostImportantAlarm(MetricsConstants mc, boolean showCameras)
 				kIterator++;
 			}
 		}
+		List<RouteSegmentResult> segments = route.getUpcomingTunnel(500);
+		if (segments != null && !segments.isEmpty()) {
+			AlarmInfo inf = new AlarmInfo(AlarmInfoType.TUNNEL, 0);
+			int d = route.getDistanceToPoint(segments.get(0).getStartPointIndex());
+			float time = speed > 0 ? d / speed : Integer.MAX_VALUE;
+			inf.setFloatValue(calculateDistance(segments, mc));
+			int vl = inf.updateDistanceAndGetPriority(time, d);
+			if (vl < value && (showCameras || inf.getType() != AlarmInfoType.SPEED_CAMERA)) {
+				mostImportant = inf;
+			}
+		}
 		return mostImportant;
 	}
 
+	private float calculateDistance(List<RouteSegmentResult> segments, MetricsConstants mc) {
```

Comment:
We have many MetricsConstants and km/m is only 1, this check is incorrect in case we will introduce MetricsConstants.METERS. The formatting is also in the wrong place should be in OsmAndFormatter